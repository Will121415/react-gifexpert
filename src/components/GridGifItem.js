import React from 'react'
import { PropTypes } from 'prop-types'
export const GridGifItem = ({url,title}) => {
  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GridGifItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}