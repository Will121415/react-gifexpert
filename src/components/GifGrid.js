import React from 'react'
import { GridGifItem } from './GridGifItem';
import { useFetchGifs } from '../Hooks/useFetchGifs';
import { PropTypes } from 'prop-types';

export const GifGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifs(category);

    // console.log(category, images, loading);


    return (
        <>
            <h3>{category}</h3>
            {loading && <strong>Loading...</strong>}
            <div className='card-grid animate__animated animate__fadeIn'>
                {
                    images.map(img => (
                        <GridGifItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>

    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}