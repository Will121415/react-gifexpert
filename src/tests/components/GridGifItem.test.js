import { GridGifItem } from "../../components/GridGifItem";
import React from "react";
import { shallow } from "enzyme";

describe('Pruba en el componente <GridGifItem />', () => {

    const title = 'Titulo del GIF';
    const url = 'https://media2.giphy.com/media/l0HlZ9q2fjX9q/200.gif';
    const wrapper = shallow(<GridGifItem  title={title} url={url}/>);


    test('Debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe tener un parrafo con el title', () => { 
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    })

    test('Debe tener la img igual al url y alt de los props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Debe tener la clase card', () => { 
        
        const div = wrapper.find('div');
        expect(div.hasClass('card')).toBe(true);
    })
})

