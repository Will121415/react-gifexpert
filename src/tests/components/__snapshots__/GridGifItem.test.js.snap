// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruba en el componente <GridGifItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Titulo del GIF"
    src="https://media2.giphy.com/media/l0HlZ9q2fjX9q/200.gif"
  />
  <p>
    Titulo del GIF
  </p>
</div>
`;
